packages/config/src/Config.ts

```diff
@@ -1,12 +1,137 @@
 import JsonFile, { JSONObject } from '@expo/json-file';
-import fs from 'fs-extra';
 import path from 'path';
 import slug from 'slugify';
 
-import { AppJSONConfig, ExpRc, ExpoConfig, PackageJSONConfig, ProjectConfig } from './Config.types';
+import {
+  AppJSONConfig,
+  ExpRc,
+  ExpoConfig,
+  PackageJSONConfig,
+  Platform,
+  ProjectConfig,
+} from './Config.types';
 import { ConfigError } from './Errors';
+import { findAndEvalConfig } from './getConfig';
+import { getRootPackageJsonPath, projectHasModule } from './Modules';
 import { getExpoSDKVersion } from './Project';
 
+/**
+ * Get all platforms that a project is currently capable of running.
+ *
+ * @param projectRoot
+ * @param exp
+ */
+function getSupportedPlatforms(
+  projectRoot: string,
+  exp: Pick<ExpoConfig, 'nodeModulesPath'>
+): Platform[] {
+  const platforms: Platform[] = [];
+  if (projectHasModule('react-native', projectRoot, exp)) {
+    platforms.push('ios', 'android');
+  }
+  if (projectHasModule('react-native-web', projectRoot, exp)) {
+    platforms.push('web');
+  }
+  return platforms;
+}
+
+export function serialize(val: any): any {
+  if (['undefined', 'string', 'boolean', 'number'].includes(typeof val)) {
+    return val;
+  } else if (typeof val === 'function') {
+    // TODO: Bacon: Should we support async methods?
+    return val();
+  } else if (Array.isArray(val)) {
+    return val.map(serialize);
+  } else if (typeof val === 'object') {
+    const output: { [key: string]: any } = {};
+    for (const property in val) {
+      if (val.hasOwnProperty(property)) {
+        output[property] = serialize(val[property]);
+      }
+    }
+    return output;
+  }
+
+  throw new Error(`Unhandled item type: ${typeof val}`);
+}
+
+/**
+ * Evaluate the config for an Expo project.
+ * If a function is exported from the `app.config.js` then a partial config will be passed as an argument.
+ * The partial config is composed from any existing app.json, and certain fields from the `package.json` like name and description.
+ *
+ * You should use the supplied `mode` option in an `app.config.js` instead of `process.env.NODE_ENV`
+ *
+ * **Example**
+ * ```js
+ * module.exports = function({ config, mode }) {
+ *   // mutate the config before returning it.
+ *   config.slug = 'new slug'
+ *   return config;
+ * }
+ *
+ * **Supports**
+ * - `app.config.js`
+ * - `app.config.json`
+ * - `app.json`
+ *
+ * @param projectRoot the root folder containing all of your application code
+ * @param options enforce criteria for a project config
+ */
+export function getConfig(
+  projectRoot: string,
+  options: {
+    mode: 'development' | 'production';
+    configPath?: string;
+    skipSDKVersionRequirement?: boolean;
+  }
+): ProjectConfig {
+  if (!['development', 'production'].includes(options.mode)) {
```

Comment:
we should document if this is impacted by toggling development mode in expo-cli

Reply:
This PR doesn't actually implement any of that. Future PRs that migrate to this config will need to account for this.